osclib/stagingapi.py

```diff
@@ -760,15 +764,27 @@ def rm_from_prj(self, project, package=None, request_id=None,
 
         self._remove_package_from_prj_pseudometa(project, package)
         subprj = self.map_ring_package_to_subject(project, package)
+        if self._supersede:
+            self.is_package_disabled(subprj, package, store=True)
         delete_package(self.apiurl, subprj, package, force=True, msg=msg)
 
         for sub_prj, sub_pkg in self.get_sub_packages(package):
             sub_prj = self.map_ring_package_to_subject(project, sub_pkg)
+            if self._supersede:
```

Comment:
you mean 'new in ring'? as we might have an existing package that is newly promoted to a ring, paired with a submission introducing new fixes needed too